rllib/algorithms/ppo/ppo_catalog.py

```diff
@@ -76,30 +76,20 @@ def __init__(
             shared=self.model_config_dict["vf_share_layers"],
         )
 
-        post_fcnet_hiddens = self.model_config_dict["post_fcnet_hiddens"]
-        post_fcnet_activation = self.model_config_dict["post_fcnet_activation"]
+        self.pi_and_vf_head_hiddens = self.model_config_dict["post_fcnet_hiddens"]
```

Comment:
Actually, here I have a question for @ArturNiederfahrenhorst :
B/c we are not knowing the framework in the constructor, we cannot really create the PiHead config yet (or at least, it'll be incomplete, not knowing its last layer size).

Shouldn't we either:
Construct the catalog already with the known framework
OR: construct a dummy distribution object here (no matter the framework) and use it to infer the action layer size.

Otherwise, we can just leave it as-is right now and construct the pi-config during the build.

Reply:
Great idea!
The initial idea here was that the same catalog could be used to pull model from any framework.
But giving it a second thought now, that's worth more or less nothing.

I think it would be fair if Catalog took a framework on instantiation. 
- Whether/how this would be used depends on the Catalog implementation.
- All "`build_some_model_or_dist`" methods would have the framework parameter removed
- We should not do the same thing in model configs though, because instances of model should be framework agnostic. As opposed to modifying RLlib with your own Catalog, where you don't pass an instance but your child class to RLlib.

I'm not a fan of a dummy distribution. That could make it so that we end up using TF code even though user uses torch or vice versa. I vote for keeping it simple and not doing such magic under the hood.